### PR TITLE
Relax Faraday version requirements

### PR DESCRIPTION
**Pull request by bdewater**
A bit of cleanup, mostly here to unlock using Faraday 2.x. See https://github.com/lostisland/faraday/blob/main/UPGRADING.md
    
Best I can tell there's no need for the middleware to follow redirects (from the JSON:API spec or the Unit API docs)

### CODE REVIEW

File: unit-ruby.gemspec [hunk 1]
```diff
@@ -35,14 +35,13 @@ Gem::Specification.new do |spec|
...
-  spec.add_dependency 'faraday_middleware', '~> 1.0.0'
+  spec.add_dependency 'faraday', '>= 1.10', '< 3'
 
-  spec.add_development_dependency 'bundler', '~> 1.17'
   spec.add_development_dependency 'dotenv', '~> 2.7.6'
   spec.add_development_dependency 'pry'
   spec.add_development_dependency 'rake', '~> 13.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rubocop', '~> 1.24.1'
+  spec.add_development_dependency 'faraday-retry'
```
Comment (ianyamey) [inline unit-ruby.gemspec:45]: Should this be in the non-dev dependencies?

```suggestion
  spec.add_dependency 'faraday-retry'
```

File: lib/unit-ruby/util/connection.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 require 'faraday'
-require 'faraday_middleware'
+begin
+  require 'faraday/retry'
+rescue LoadError
+  puts "Add `gem 'faraday-retry'` to your Gemfile"
+  raise
+end
```
Comment (ianyamey) [inline lib/unit-ruby/util/connection.rb:7]: From what I can tell, `faraday-retry` requires `faraday` version `~> 2.0`. Do you think we might need to do something to require either 'faraday/retry' or 'faraday-middleware' depending on the faraday version?

**Review by ianyamey — COMMENTED** [review]
Thanks for the contribution!

**Comment by bdewater** [comment]
My bad, that was a bit of a leftover from when I was trying to make faraday-middleware work (which can't be used with faraday 2.0) before I decided to bump the requirement to at least 1.10 and make this easier :)

`faraday` 1.10.x [requires](https://rubygems.org/gems/faraday/versions/1.10.2) `faraday-retry` version `~> 1.0`, you are correct about `faraday-retry` 2.x, so it should just work without a version constraint specified. I amended the last commit.

**Review by ianyamey — APPROVED** [review]

**Comment by ianyamey** [comment]
Closing as duplicate of #29

**Pull request closed without merging**